packages/components/src/components/KeyValueList/KeyValueList.js

```diff
@@ -0,0 +1,143 @@
+  const keyValueFields = keyValues.map(
+    ({ id, key, keyPlaceholder, value, valuePlaceholder }, index) => {
+      const keyId = `${id}-key`;
+      const valueId = `${id}-value`;
+      const invalidKey = keyId in invalidFields;
+      const invalidValue = valueId in invalidFields;
+      invalid = invalid || invalidKey || invalidValue;
+      return (
+        <div className="keyvalueRow" key={`keyvalueRow${id}`}>
+          <TextInput
+            id={keyId}
+            labelText=""
+            aria-label={intl.formatMessage(
+              {
+                id: 'dashboard.keyValueList.keyTextInput.ariaLabel',
+                defaultMessage: '{legendText} Key#{index}. {ariaLabelKey}'
+              },
+              { legendText, index, ariaLabelKey }
+            )}
+            value={key}
+            placeholder={keyPlaceholder}
+            onChange={e => {
+              onChange({ type: 'key', index, value: e.target.value });
+            }}
+            invalid={invalidKey}
+            autoComplete="off"
+          />
+          <div className="colon">:</div>
+          <TextInput
+            id={valueId}
+            labelText=""
+            aria-label={intl.formatMessage(
+              {
+                id: 'dashboard.keyValueList.valueTextInput.ariaLabel',
+                defaultMessage: '{legendText} Value#{index}. {ariaLabelValue}'
+              },
+              { legendText, index, ariaLabelValue }
+            )}
+            value={value}
+            placeholder={valuePlaceholder}
+            onChange={e => {
+              onChange({ type: 'value', index, value: e.target.value });
+            }}
+            invalid={invalidValue}
+            autoComplete="off"
+          />
+          {deleteButton && value && (
+            <Delete
+              className="deleteInputIcon"
+              aria-label={intl.formatMessage(
+                {
+                  id: 'dashboard.keyValueList.deleteButton.ariaLabel',
+                  defaultMessage:
+                    'Delete Button#{index}. Deletes entire entry in current text field.'
+                },
+                { index }
+              )}
+              onClick={() => {
+                onChange({ type: 'value', index, value: '' });
+              }}
+            />
+          )}
+        </div>
+      );
+    }
+  );
+
+  return (
+    <div className="keyvalues">
+      <div className="labelAndButtons">
+        <p className="label">{legendText}</p>
+        <Remove
+          className={
+            keyValueFields.length === minKeyValues
+              ? 'removeIconDisabled'
+              : 'removeIcon'
+          }
+          onClick={onRemove}
+          aria-label={intl.formatMessage(
+            {
+              id: 'dashboard.keyValueList.removeButton.ariaLabel',
+              defaultMessage: 'Remove Button. Removes {legendText} entry.'
+            },
+            { legendText }
+          )}
+        />
+        <Add
```

Comment:
Sorry I didn't pick up on this in previous reviews, I didn't realise these were just icons and not buttons.
We should use Button components here with the icon prop set, this will ensure keyboard accessibility etc. works as expected. If you want to hide the button border and background you can set `kind="ghosh"`. See LabelFilter, LogoutButton, or Rerun for examples.